addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -106,10 +106,27 @@ def _timesheet_prepare_line_values(self, index, work_hours_data, day_date, work_
             'company_id': self.holiday_status_id.timesheet_task_id.company_id.id or self.holiday_status_id.timesheet_project_id.company_id.id,
         }
 
+    def _check_missing_global_leave_timesheets(self):
+        if not self:
+            return
```

Comment:
It seems weird to me to see we have to do that. 🤔 
Do you know in which case we could have no record in self? 🤔

Reply:
In my case, it happened with the 13th Month tests of `test_l10n_be_hr_payroll_account`. When going through [this function](https://github.com/odoo/odoo/blob/15.0/addons/hr_work_entry_holidays/models/hr_work_entry.py#L26), if none of the work entries have a `leave_id` linked, `action_refuse()` will be called with an empty recordset.